### PR TITLE
feat: So a User can See when an update is available and starts downloading

### PR DESCRIPTION
**Pull request by thereaper90917**
Notification:  for When Updates Are Available the logging is cool but from a users perspective it would be nice to see that a 

download has started especially if you have slow internet.

### CODE REVIEW

**Review by malept — CHANGES_REQUESTED** [review]

File: index.js [hunk 1]
```diff
@@ -64,6 +64,12 @@ function initUpdater (opts) {
 
   autoUpdater.on('update-available', () => {
     log('update-available; downloading...')
+     // Creates Notification For User When a Download is Available
+     const notification = {
+      body: 'Update Available Downloading..',
+      timeoutType: "10"
```
[the remark this answers is not in the document]
Comment (thereaper90917) [reply index.js]: added. the requested parameters to make the process more dynamic you can now, choose the message you want to show. and how long the notification should be also, you can enable or disable the notification.

**Review by Aivolt1 — APPROVED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -32,6 +32,19 @@ declare namespace updateElectronApp {
...
+      readonly userNotificationMessage?: string;
+      /**
+       * @param {String} userNotificationTime Sets Time For How Long User Notification is shown. Defaults to !0 Seconds.
+       */
+       readonly userNotificationTime?: string;
+       /**
+        * @param {Boolean} userNotification Defaults to False. when enabled a User will be notified when a update download starts.
+        */
+        readonly userNotification?: boolean;
+
```
Comment (realDragonium) [inline index.d.ts:47]: Fix indents please

File: index.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ const userAgent = format(
   os.platform(),
   os.arch()
 )
+
```
Comment (realDragonium) [inline index.js:18]: remove please

File: index.js [hunk 4]
```diff
@@ -33,9 +35,11 @@ module.exports = function updater (opts = {}) {
     : opts.electron.app.on('ready', () => initUpdater(opts))
 }
 
+
+
```
Comment (realDragonium) [inline index.js:39]: unnecessary

File: index.js [hunk 5]
```diff
@@ -33,9 +35,11 @@ module.exports = function updater (opts = {}) {
...
 function initUpdater (opts) {
-  const { host, repo, updateInterval, logger, electron } = opts
-  const { app, autoUpdater, dialog } = electron
+  const { host, repo, updateInterval, logger, electron, userNotificationMessage, userNotificationTime} = opts
```
Comment (realDragonium) [inline index.js:41]: Missing a space before the `}`

File: index.js [hunk 6]
```diff
@@ -64,6 +68,15 @@ function initUpdater (opts) {
...
     log('update-available; downloading...')
+     // Creates Notification For User When a Download is Available
+     if(opts.userNotification){
+      const notification = {
+        body: userNotificationMessage,
+        timeoutType: userNotificationTime
+      }
+       new Notification(notification).show()
+     }
+
```
Comment (realDragonium) [inline index.js:79]: unnecessary new line

File: index.js [hunk 7]
```diff
@@ -98,13 +111,19 @@ function validateInput (opts) {
...
     logger: console,
-    notifyUser: true
+    notifyUser: true,
+    userNotification:false,
+    userNotificationMessage:"Updating..",
+    userNotificationTime:"10"
   }
-  const { host, updateInterval, logger, notifyUser } = Object.assign({}, defaults, opts)
+  const { host, updateInterval, logger, notifyUser,userNotification, userNotificationMessage, userNotificationTime,  } = Object.assign({}, defaults, opts)
+
```
Comment (realDragonium) [inline index.js:120]: unnecessary newline

File: index.js [hunk 8]
```diff
@@ -98,13 +111,19 @@ function validateInput (opts) {
...
   }
-  const { host, updateInterval, logger, notifyUser } = Object.assign({}, defaults, opts)
+  const { host, updateInterval, logger, notifyUser,userNotification, userNotificationMessage, userNotificationTime,  } = Object.assign({}, defaults, opts)
+  
 
   // allows electron to be mocked in tests
   const electron = opts.electron || require('electron')
 
+
+
```
Comment (realDragonium) [inline index.js:126]: unnecessary newlines

**Review by realDragonium — CHANGES_REQUESTED** [review]
Please change those things, thanks :slightly_smiling_face:

File: index.d.ts (hunk 2 above)
Comment (thereaper90917) [reply index.d.ts]: fixed indents

File: index.js (hunk 4 above)
Comment (thereaper90917) [reply index.js]: fixed

File: index.js (hunk 3 above)
Comment (thereaper90917) [reply index.js:18]: fixed

File: index.js (hunk 7 above)
Comment (thereaper90917) [reply index.js]: removed

File: index.js (hunk 8 above)
Comment (thereaper90917) [reply index.js]: removed

File: index.js (hunk 6 above)
Comment (thereaper90917) [reply index.js]: removed

File: index.js (hunk 5 above)
Comment (thereaper90917) [reply index.js]: added

**Review by realDragonium — APPROVED** [review]
Looks a lot better now, thanks!

**Review by DJRolls — APPROVED** [review]
I love

**Review by erickzhao — CHANGES_REQUESTED** [review]
Hey @thereaper90917, thanks for the PR. I don't think notifying users on download is something we want to explicitly support since it can be disruptive to users.

Maybe an alternative would be to allow a callback for Electron apps to hook their own code once the download is started?

**Pull request closed without merging**